lxd/device/disk.go

```diff
@@ -345,6 +345,43 @@ func (d *disk) validateConfig(instConf instance.ConfigReader) error {
 		}
 	}
 
+	// Extract initial configuration from the profile and validate them against appropriate storage driver.
+	// Currently initial configuration is only applicable to root disk devices.
+
+	initialConfig := make(map[string]string)
+	for k, v := range d.config {
+		if strings.Contains(k, "initial.") {
+			newKey := strings.TrimPrefix(k, "initial.")
+			initialConfig[newKey] = v
+		}
+	}
+
+	if len(initialConfig) > 0 {
+		if !shared.IsRootDiskDevice(d.config) {
+			return fmt.Errorf("Non-root disk device cannot contain initial.* configuration")
+		}
+
+		d.pool, err = storagePools.LoadByName(d.state, d.config["pool"])
+		if err != nil {
+			return err
+		}
+
+		// Create temporary volume definition.
+		vol := storageDrivers.NewVolume(
+			d.pool.Driver(),
+			d.pool.Name(),
+			storageDrivers.VolumeTypeImage,
```

Comment:
This is the wrong volume type, it should be either storageDrivers.VolumeTypeContainer or storageDrivers.VolumeTypeVM based on the instance type (which is another reason to do it only when we have an d.inst set).